### PR TITLE
Fix services image refresh

### LINKED ISSUE #1755: Image refresh for services
Image refresh for services
 * [ ] FAIL: image-refresh services

### PR DESCRIPTION
**Pull request by martinpitt**
This is some larger reorgnization, I really coudln't find a way to split this up sensibly. Details in the individual commits. This *is* an API break, we'll need to adjust the tests -- but I think it's ultimately more future proof now.

Fixes #1755

 * [x] image-refresh services
 * [x] Adjust cockpit tests: https://github.com/cockpit-project/cockpit/pull/15542 (approved, will land in lockstep)
 * [x] Adjust subscription-manager tests: https://github.com/candlepin/subscription-manager/pull/2504

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-87s4d.
Log: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-1768-20210310-061414/

**Comment by cockpituous** [comment]
Task failed: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-1768-20210310-061414/

**Comment by martinpitt** [comment]
Awesome, the new freeipa container actually seems broken :/ Needs work.

**Comment by martinpitt** [comment]
There is no available container image that works on a centos-7 host. I sent https://github.com/freeipa/freeipa-container/issues/383 about that with my investigations. Until then, this is blocked.

**Comment by martinpitt** [comment]
This is really awkward. [candlepin](https://github.com/candlepin/ansible-role-candlepin) *still* is not ported to CentOS/RHEL 8, and the playbooks are totally broken there. There is also no container image for it. OTOH freeipa containers are now broken on CentOS 7. So, rock ←→ hard place :-(

I wasted 3 hours on that already, I need to move to something else. Blocked for now.

File: images/scripts/services.setup [hunk 1]
```diff
@@ -43,7 +44,7 @@ docker run -d --rm --name freeipa -ti -h f0.cockpit.lan \
     -p 53:53/udp -p 53:53 -p 80:80 -p 443:443 -p 389:389 -p 636:636 -p 88:88 -p 464:464 -p 88:88/udp -p 464:464/udp -p 123:123/udp \
     -v /var/lib/ipa-data:/data:Z \
     -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
-    freeipa/freeipa-server:fedora-rawhide \
+    quay.io/freeipa/freeipa-server:fedora-rawhide \
```
Comment (mvollmer) [inline images/scripts/services.setup:47]: The comment above "docker pull quay.io/freeipa/freeipa-server:fedora-rawhide" should probably also be updated to mention quay.io.

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-1768-20210316-140101/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1768-20210316-140101/

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-1768-20210316-175959/

**Comment by cockpituous** [comment]
image-refresh services done: https://github.com/cockpit-project/bots/commits/image-refresh-services-20210316-183131

File: images/scripts/services.setup (hunk 1 above)
Comment (martinpitt) [reply images/scripts/services.setup:37]: Done.

**Comment by martinpitt** [comment]
Running tests in https://github.com/cockpit-project/cockpit/pull/15542

**Comment by martinpitt** [comment]
Sent port for subscription-manager, testing in https://github.com/candlepin/subscription-manager/pull/2504

**Comment by martinpitt** [comment]
https://github.com/cockpit-project/bots/pull/1768 landed, so this is ready to land now in lockstep with https://github.com/cockpit-project/cockpit/pull/15542 . After it lands, I'll do the backport for the rhel-* branches.

**Review by marusak — APPROVED** [review]
Thanks!

**Pull request merged**